DS-12169 WAE public beta (#5924)
cloudflare/cloudflare-docs: content/analytics/analytics-engine/sql-reference.md

---
title: SQL Reference
pcx_content_type: reference
weight: 5
meta:
  title: Workers Analytics Engine SQL Reference
---

# Workers Analytics Engine SQL Reference

## SHOW statement

> `SHOW` can be used to list the tables on your account. The table name is the name you specified as `dataset` when configuring the workers binding (see [Get started with Workers Analytics Engine](../get-started/#1-configure-your-dataset-and-binding-in-wrangler)). The table is automatically created when you write event data in your worker. 

Reviewer: ```suggestion
`SHOW` can be used to list the tables on your account. The table name is the name you specified as `dataset` when configuring the workers binding (refer to [Get started with Workers Analytics Engine](../get-started/#1-configure-your-dataset-and-binding-in-wrangler), for more information). The table is automatically created when you write event data in your worker. 
```

```SQL
SHOW TABLES
[FORMAT <format>]
```

> See [below](#format-clause) for the available `FORMAT` options.

Reviewer: ```suggestion
Refer to [FORMAT clause](#format-clause) for the available `FORMAT` options.
```

## SELECT statement

`SELECT` is used to query tables.

Usage:
```SQL
SELECT <expression_list>
[FROM <table>|(<subquery>)]
[WHERE <expression>]
[GROUP BY <expression>, ...]
[ORDER BY <expression_list>] 
[LIMIT <n>|ALL]
[FORMAT <format>]
```

> See below for the syntax of each clause and see the [SQL API docs](../sql-api/) for some example queries.

Reviewer: ```suggestion
Below you can find the syntax of each clause. Refer to the [SQL API docs](../sql-api/) for some example queries.
```

```JSON
{"<column 1 name>": "<column 1 value>", "<column 2 name>": "<column 2 value>"}
{"<column 1 name>": "<column 1 value>", "<column 2 name>": "<column 2 value>"}
...
```

#### TabSeparated

Data is returned with newline separated rows. Columns are separated with tabs. There is no header.

```TSV
column 1 value  column 2 value
column 1 value  column 2 value
...
```

## Supported functions

### COUNT

Usage:
```SQL
COUNT()
```

Count is an aggregation function that returns the number of rows in each group or results set.

### SUM

Usage:
```SQL
SUM(column_name)
```

Sum is an aggregation function that returns the sum of column values across all rows in each group or results set.

Example:
```SQL
-- return the total cost of all items
SUM(item_cost)
```

### AVG

Usage:
```SQL
AVG(column_name)
```

Avg is an aggregation function that returns the mean of column values across all rows in each group or results set.

Example:
```SQL
-- return the mean item cost
AVG(item_cost)
```

### QUANTILEWEIGHTED

Usage:
```SQL
QUANTILEWEIGHTED(q, column_name, weight_column_name) 
```

> `QUANTILEWEIGHTED` is an aggregation function that returns the value at the q<sup>th</sup> quantile in the named column across all rows in each group or results set. Each row will be weighted by the value in `weight_column_name`. Typically this would be `_sample_interval` (see (how sampling works)[../sql-api/#sampling]).

Reviewer: ```suggestion
`QUANTILEWEIGHTED` is an aggregation function that returns the value at the q<sup>th</sup> quantile in the named column across all rows in each group or results set. Each row will be weighted by the value in `weight_column_name`. Typically this would be `_sample_interval` (refer to [how sampling works](../sql-api/#sampling), for more information).
```

Example:
```SQL
-- estimate the median value of <double1>
QUANTILEWEIGHTED(0.5, double1, _sample_interval) 

-- in a table of query times, estimate the 95th centile query time
QUANTILEWEIGHTED(0.95, query_time, _sample_interval)
```

### IF

Usage:
```SQL
IF(<condition>, <true_expression>, <false_expression>)
```

Returns `<true_expression>` if `<condition>` evaluates to true, else returns `<false_expression>`.

Example:
```SQL
IF(temp > 20, 'It is warm', 'Bring a jumper')
```

### INTDIV

Usage:
```SQL
INTDIV(a, b)
```

> Divide a by b rounding the answer down to the nearest whole number.

Reviewer: ```suggestion
Divide a by b, rounding the answer down to the nearest whole number.
```

### TOUINT32

Usage:
```SQL
TOUINT32(<expression>)
```

Converts any numeric expression, or expression resulting in a string representation of a decimal, into an unsigned 32 bit integer.

Behaviour for negative numbers is undefined.

### TODATETIME

Usage:
```SQL
TODATETIME(<expression>)
```

`TODATETIME` converts an expression to a datetime.

Examples:
```SQL
-- double1 contains a unix timestamp in seconds
TODATETIME(double1)

-- blob1 contains an datetime in the format 'YYYY-MM-DD hh:mm:ss'
TODATETIME(blob1)

-- literal values:
TODATETIME(355924804) -- unix timestamp
TODATETIME('355924804') -- string containing unix timestamp
TODATETIME('1981-04-12 12:00:04') -- string with datetime in 'YYYY-MM-DD hh:mm:ss' format
```

### NOW

Usage:

> ```

Reviewer: ```suggestion
```SQL
```
